### PR TITLE
Respond with phydat as CBOR

### LINKED ISSUE #12: Submit values via a defined cbor interface
We currently only append the sensor values to the payload of our response. We should define a data structure / format that gets used by this package. This data structure should also contain an info about the scaling of the values. (Recommended library: tinycbor.)

### PR DESCRIPTION
**Pull request by rosetree**
This pull request changes the `_sense_type_handler` to respond with CBOR instead of text. Affects all resources by that handler (`/sensor`, `/temp`, `/hum` …). The data format is documented in `README.md` (and the presentation for milestone 3 in #20). in the response are also all available values of a single device, thus resolving part of #8. @EM000 might be interested in this change, because the web app needs to handle the response differently.

Closes #12

### CODE REVIEW

File: saul_coap.c [hunk 1]
```diff
@@ -198,15 +204,86 @@ static ssize_t _saul_type_handler(coap_pkt_t* pdu, uint8_t *buf, size_t len, voi
...
+{
+    CborEncoder mapEncoder, aryEncoder;
+    CborError err = CborNoError;
+
+    cbor_encoder_init(encoder, cbor_buf, buf_len, 0);
+    if (err != CborNoError) {
+        return err;
+    }
+
+    err = cbor_encoder_create_map(encoder, &mapEncoder, 3);
```
Comment (rosetree) [inline saul_coap.c:232]: We should use `PHYDAT_DIM` instead of magic `3`.

Comment (rosetree) [reply saul_coap.c:232]: Nope. The dimension is not used in the map, but in the values array. And we don’t use `PHYDAY_DIM` there, but `dim` – the actual available dimensions.

File: README.md [hunk 2]
```diff
@@ -36,6 +40,42 @@ The idea of these resources is, to offer similar functionality as the
 - `/saul/dev` (POST) needs an ID as argument. Returns some information
   about the sensor for that ID (name and type). 
 
+## Phydat as CBOR
```
Comment (MatthiasBraeuer) [inline README.md]: Please add a short explanation what CBOR is.

File: saul_coap.c [hunk 3]
```diff
@@ -198,15 +204,86 @@ static ssize_t _saul_type_handler(coap_pkt_t* pdu, uint8_t *buf, size_t len, voi
...
+    }
+
+    err = cbor_encode_int(&mapEncoder, data.scale);
+    if (err != CborNoError) {
+        return err;
+    }
+
+    err = cbor_encoder_close_container(encoder, &mapEncoder);
+    if (err != CborNoError) {
+	return err;
```
Comment (MatthiasBraeuer) [inline saul_coap.c]: Please adjust the indentation

**Review by MatthiasBraeuer — CHANGES_REQUESTED** [review]

File: saul_coap.c (hunk 3 above)
Comment (rosetree) [reply saul_coap.c]: Fixed it.

File: README.md (hunk 2 above)
Comment (rosetree) [reply README.md]: Please see 643319649963a74d569d6dd6ed96f18e6b287ec4; do you think this is enough?

**Pull request merged**